src/rest/assets/client.ts

```diff
@@ -11,11 +11,17 @@ import {
 } from '~/rest/assets/params';
 import { RestClient } from '~/rest/client';
 import { TxBase } from '~/rest/common';
-import { PostResult } from '~/rest/interfaces';
+import { PostResult, RestSuccessResult } from '~/rest/interfaces';
 
 export class Assets {
   constructor(private client: RestClient) {}
 
+  public async createAndGetAssetId(params: ReturnType<typeof createAssetParams>): Promise<string> {
+    const result = (await this.createAsset(params)) as RestSuccessResult;
+
+    return result.asset as string;
+  }
+
```

Comment:
Even if we add in the PostResult<{ asset: string }> extended version, we will have to assert at all places that the result is RestSuccessResult (not RestErrorResult) and then get the asset from the result part. I wanted to avoid asserting this at all places so felt this should be ok to add in here. I can move this to utils if that makes more sense